http4s-blaze-server/src/main/scala/com/avast/server/toolkit/http4s/CorrelationIdMiddleware.scala

```diff
@@ -0,0 +1,61 @@
+package com.avast.server.toolkit.http4s
+
+import java.util.UUID
+
+import cats.data.{Kleisli, OptionT}
+import cats.effect.Sync
+import io.chrisdavenport.vault.Key
+import org.http4s.util.CaseInsensitiveString
+import org.http4s.{Header, HttpRoutes, Request, Response}
+import org.slf4j.LoggerFactory
+
+import scala.language.higherKinds
+
+/** Provides correlation ID functionality. Either generates new correlation ID for a request or takes the one sent in HTTP header
+  * and puts it to [[org.http4s.Request]] attributes. It is also filled into HTTP response header.
+  */
+class CorrelationIdMiddleware[F[_]: Sync](correlationIdHeaderName: CaseInsensitiveString, generator: () => String) {
```

Comment:
can this be a trait?

Reply:
why would you?

Reply:
It could but I don't think we need to abstract away middleware.